### PR TITLE
Issue #998: migrate CLI commands

### PR DESCRIPTION
**Pull request by Vinai**
This is the first of the CLI commands to be moved back to the catalog repository (from sample-project).  
Also refactored to make testable and test coverage.

### CODE REVIEW

File: src/ConsoleCommand/CliBootstrap.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    const ENV_DEBUG_VAR = 'LP_DEBUG_LOG';
+
+    public static function create(string $cliCommandClass, Factory ...$factoriesToRegister): BaseCliCommand
+    {
+        $masterFactory = self::createMasterFactory($factoriesToRegister);
+
+        return new $cliCommandClass($masterFactory, new CLImate());
+    }
+
+    private static function createMasterFactory(array $factoriesToRegister): MasterFactory
```
Comment (tim-bezhashvyly) [inline src/ConsoleCommand/CliBootstrap.php]: Should have either PHPDock or be variadic.

File: src/ConsoleCommand/CliFactoryBootstrap.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+{
+    protected static $projectFactoryClass = ProjectFactory::class;
+
+    protected static $commonFactoryClass = CommonFactory::class;
+
+    protected static $loggingFactories = [
+        LoggingDomainEventHandlerFactory::class,
+        LoggingCommandHandlerFactory::class,
+        LoggingQueueFactory::class,
+    ];
```
Comment (tim-bezhashvyly) [inline src/ConsoleCommand/CliFactoryBootstrap.php]: Why 3 properties above are protected?

File: src/ConsoleCommand/CliFactoryBootstrap.php [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+    public static function createMasterFactory(Factory ...$factoriesToRegister): MasterFactory
+    {
+        $masterFactory = new \LizardsAndPumpkins\Util\Factory\SampleMasterFactory();
```
Comment (tim-bezhashvyly) [inline src/ConsoleCommand/CliFactoryBootstrap.php]: Please import.

File: src/ConsoleCommand/CliFactoryBootstrap.php [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+     */
+    private static function getDefaultFactoryClasses(): array
+    {
+        return [
+            static::$commonFactoryClass,
+            static::$projectFactoryClass,
+        ];
+    }
+
+    private static function arrayContainsInstanceOfClass(array $objects, string $class): bool
```
Comment (tim-bezhashvyly) [inline src/ConsoleCommand/CliFactoryBootstrap.php]: Missing PHPDoc.

File: src/ConsoleCommand/Command/RunImport.php [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+    {
+        $this->factory = $factory;
+        $this->setCLImate($CLImate);
+    }
+    
+    /**
+     * @param CLImate $climate
+     * @return array[]
+     */
+    protected function getCommandLineArgumentsArray(CLImate $climate) : array
```
Comment (tim-bezhashvyly) [inline src/ConsoleCommand/Command/RunImport.php]: Maybe final?

File: src/ConsoleCommand/Command/RunImport.php [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+                ],
+                'importFile'    => [
+                    'description' => 'Import XML file',
+                    'required'    => true,
+                ],
+            ]
+        );
+    }
+
+    protected function execute(CLImate $CLImate)
```
Comment (tim-bezhashvyly) [inline src/ConsoleCommand/Command/RunImport.php]: Maybe final?

File: src/ConsoleCommand/CliFactoryBootstrap.php (hunk 2 above)
Comment (Vinai) [inline src/ConsoleCommand/CliFactoryBootstrap.php]: These properties are protected for testability purposes. Do you want me to use protected methods instead?

Comment (tim-bezhashvyly) [inline src/ConsoleCommand/CliFactoryBootstrap.php]: No, that's fine.

**Comment by Vinai** [comment]
Ready for next review.

**Pull request merged**